chips/nrf52/src/ieee802154_radio.rs

```diff
@@ -1035,11 +1170,18 @@ impl<'a> kernel::hil::radio::RadioConfig<'a> for Radio<'a> {
     }
     fn stop(&self) -> Result<(), ErrorCode> {
         self.radio_off();
+        self.state.set(RadioState::OFF);
         Ok(())
     }
     fn is_on(&self) -> bool {
-        true
+        if self.registers.power.is_set(Task::ENABLE) {
+            true
+        } else {
+            false
+        }
     }
+
+    // Previous driver implementation //
```

Comment:
can this fn just be deleted?

Reply:
No, this fn is part of the RadioConfig trait (see below)

`pub trait RadioConfig<'a> {
    /// buf must be at least MAX_BUF_SIZE in length, and
    /// reg_read and reg_write must be 2 bytes.
    fn initialize(
        &self,
        spi_buf: &'static mut [u8],
        reg_write: &'static mut [u8],
        reg_read: &'static mut [u8],
    ) -> Result<(), ErrorCode>;
    fn reset(&self) -> Result<(), ErrorCode>;
    fn start(&self) -> Result<(), ErrorCode>;
    fn stop(&self) -> Result<(), ErrorCode>;
    fn is_on(&self) -> bool;
    fn busy(&self) -> bool;

    fn set_power_client(&self, client: &'a dyn PowerClient);

    /// Commit the config calls to hardware, changing the address,
    /// PAN ID, TX power, and channel to the specified values, issues
    /// a callback to the config client when done.
    fn config_commit(&self);
    fn set_config_client(&self, client: &'a dyn ConfigClient);

    fn get_address(&self) -> u16; //....... The local 16-bit address
    fn get_address_long(&self) -> [u8; 8]; // 64-bit address
    fn get_pan(&self) -> u16; //........... The 16-bit PAN ID
    fn get_tx_power(&self) -> i8; //....... The transmit power, in dBm
    fn get_channel(&self) -> u8; // ....... The 802.15.4 channel

    fn set_address(&self, addr: u16);
    fn set_address_long(&self, addr: [u8; 8]);
    fn set_pan(&self, id: u16);
    fn set_tx_power(&self, power: i8) -> Result<(), ErrorCode>;
    fn set_channel(&self, chan: u8) -> Result<(), ErrorCode>;
}`

Reply:
Yeah, it looks like that's what the rf233 does.

The radio TRD says:
> The `busy` function returns whether the radio is currently busy.
It MUST return false if the radio is currently idle and can accept
reconfiguration or packet transmission requests. If it is busy and
cannot accept reconfiguration or packet transmission requests, it
MUST return true.

Would be good to make this an honest implementation eventually, but that can wait until a future PR.